CAN-D/Src/freertos.c

```diff
@@ -31,25 +35,57 @@ void MX_FREERTOS_Init(void) {
 
   /* Create the thread(s) */
   /* definition and creation of defaultTask */
-  osThreadDef(defaultTask, StartDefaultTask, osPriorityNormal, 0, 128);
-  defaultTaskHandle = osThreadCreate(osThread(defaultTask), NULL);
+  osThreadDef(canTask, canBridgeTask, osPriorityNormal, 0, 128);
+  canTaskHandle = osThreadCreate(osThread(canTask), NULL);
 
   /* add threads, ... */
 
   /* add queues, ... */
-
+  osMessageQDef(canDataQueue, 8, uint8_t);
+  canDataQueueHandle = osMessageCreate(osMessageQ(canDataQueue), NULL);
 }
 
 /**
-  * @brief  Function implementing the defaultTask thread.
+  * @brief  Function implementing the canBridgeTask thread.
   * @param  argument: Not used 
   * @retval None
   */
-/* USER CODE END Header_StartDefaultTask */
-void StartDefaultTask(void const * argument)
+void canBridgeTask(void const * argument)

```

Comment:
We should almost always queue the data to the CAN and just control if interrupts are enabled/disabled in order to determine if we are listening to the can bus. Thoughts?

Reply:
I'm a little unsure what you mean by "queue the data to the CAN". I think what you're saying is fine though. We'll still use the bridge to control what we're doing with the CAN data right? So that we can SPI transfer it to the SD, or stream it wirelessly, or send to the PC via USB?

Reply:
I should have said, "queue the data from can". Maybe I don't fully understand what you are doing here. Is the flow: Incoming CAN data -> Read in the can.c file -> enqueue rx data -> dequeue data here -> ???

Or am I not understanding this properly?

Reply:
Yes, that's it! One little thing though is that the CAN data isn't "incoming" because it's already in the CAN FIFO. Then HAL_CAN_GetRxMessage() just loads the data into a buffer.